src/Compilers/Core/Portable/Operations/OperationNodes.cs

```diff
@@ -379,6 +457,30 @@ protected override (bool hasNext, int nextSlot, int nextIndex) MoveNext(int prev
             }
         }
 
+        internal override (bool hasNext, int nextSlot, int nextIndex) MoveNextReversed(int previousSlot, int previousIndex)
+        {
+            switch (previousSlot)
+            {
+                case int.MaxValue:
+                    if (Arguments.IsEmpty) goto case 1;
+                    else return (true, 1, Arguments.Length - 1);
```

Comment:
Consider using `if (!Arguments.IsEmpty) ...` to match the style of the other implementations.